### ISSUE TITLE
Add option to function generator to support Netlify background functions

### ISSUE THREAD

**Issue by dthyresson** [redwoodjs/redwood#1413]
Netlify has added support for background functions.

> if you name a function

`hello.js`

>It will deploy as a synchronous function that returns its response to the client after the function completes. Because the client is waiting on a response, the function must finish within 10 seconds, or it will fail.

>However, if you change the name to

`hello-background.js`

> It will deploy as a background function that returns an immediate client response on successful invocation, then continues running in the background until it completes asynchronously — up to 15 minutes later.

See: https://www.netlify.com/blog/2020/10/29/announcing-background-functions/

Should RW add an option like `--background` to name the function according to this convention.

Of course, this case is Netlify-specific, but perhaps one can adopt this naming convention even when deployed to AWS Serverless, etc?

**Comment by dthyresson** [comment]
Also,

> perform-background

becomes camelized to `performBackground"

```
yarn rw g function perform-background                    

yarn run v1.22.10
$ '/.../node_modules/.bin/rw' g function perform-background
  ✔ Generating function files...
    ✔ Successfully wrote file `./api/src/functions/performBackground.js`
✨  Done in 1.17s.
```

**Comment by thedavidprice** [comment]
I'm assuming that we'll see other providers add background job functionality as well. Without knowing how they'll handle config/convention, I +1 any implementation that feels general enough to allow for a possible extension to other providers. 

For your idea here, DT, that might mean the option should be named `--background-netlify`. (maybe?)

Looking forward to seeing more of these types of infrastructure features coming online!!

**Comment by dthyresson** [comment]
What if the command line options was not to "background it" or `--background-netlify` but to "suffix" it?

`--suffix background`

?

That does not see worth it though.

What id a `--background` flag checked for the deployment target and transformed the filename accordingly?

Currently, the only target would be `netlify` and it would add a background suffix, and for others it would with camelcase it still as `performBackground` or not transform it `perform`?

A Netlify target would check for the existence of a `netlify toml` config file.

**Comment by dthyresson** [comment]
Closing in favor of people creating their own generator templates. This idea is a slippery slope: Webhooks flavors, background Netlify functions, Vercel background functions, Netlify builder functions, etc etc etc.
